source/slang/core.meta.slang

```diff
@@ -1040,16 +1040,15 @@ __generic<T, let N:int>
 __magic_type(ArrayExpressionType)
 struct Array : IArray<T>
 {
-    [ForceInline]
-    int getCount() { return N; }
+    __intrinsic_op($(kIROp_GetArrayLength))
```

Comment:
It looks like this member-function should be labeled `[__readNone]`

Reply:
`[__readNone]` is only meaningful for ordinary functions with a body. Its purpose is to make DCE understand that a `IRCall` to the function is side-effect free. Since this function has an `__intrinsic_op` that maps it to an instruction during IR lowering, there will be no `IRCalls` to it, and there is no need for the attribute. We already added a case in `mightHaveSideEffects` for the new opcode, and that should be sufficient for DCE purposes.